splunklogger.js

```diff
@@ -513,17 +699,35 @@ SplunkLogger.prototype.flush = function (callback) {
     // After running all, if any, middlewares send the events
     var that = this;
     utils.chain(callbacks, function(err, passedContext) {
+        // The passedContext parameter could be named context to
+        // do this automatically, but the || notation adds a bit of clarity.
+        context = passedContext || context;
+
         // Errors from any of the middleware callbacks will fall through to here
         // If the context is modified at any point the error callback will get it also
         // event if next("error"); is called w/o the context parameter!
         // This works because context inside & outside the scope of this function
         // point to the same memory block.
-        // The passedContext parameter could be named context to
-        // do this automatically, but the || notation adds a bit of clarity.
         if (err) {
-            that.error(err, passedContext || context);
+            that.error(err, context);
         }
         else {
+            // Validate the context again, right before using it
+            context = that._initializeContext(context);
+            context.requestOptions.headers["Authorization"] = "Splunk " + context.config.token;
+
+            if (isBatched) {
+                // Don't run _makeBody since we've already done that for batching
+                // Manually set the content-type header for batched requests, default is application/json
+                // since json is set to true.
+                context.requestOptions.headers["content-type"] = "application/x-www-form-urlencoded";
```

Comment:
Not sure why the JSON content-type doesn't work, but that header is force by the option `json=true`. Another way to do this would be committing the `json=true` request option and manually calling `JSON.parse()` on every response.